design/specs/table_inspector.md

```diff
@@ -0,0 +1,163 @@
+
+To disable the table inspector:
+
+- Directly click on the inspector toggle located in the table toolbar to hide the table inspector.
+
+## Using the Table Inspector
+
+Whatever is presently selected will determine the content of the table inspector. During non-selection mode, the inspector simply displays properties for the current table.
+
+### Table Inspector Modes
+
+The current selection is what triggers the various inspection modes. The inspector panel's header contains a brief description of what is included in the selection.
+
+### Table Properties
+
+Shown during non-selection mode.
+
+Table properties include:
+
+- Table Name and Description
+- Table Links
+- Table Display Options
+  - Record Summary
+  
+![image](/assets/design/specs/table_inspector/182566720-3622f073-24ee-4aff-8460-64442f271fea.png)
+
+### Column Properties
+
+Shown when a [column](#column-selection) or [cell](#cell-selection) are selected.
+
+Column properties include:
+
+- Column Name
+- Column Constraints
+  - Allow NULL
+  - Allow Duplicates
+- Column Data Type
+  - Select Type
+  - Database Options
+  - Format Options
+  
+![image](/assets/design/specs/table_inspector/182350473-c20ebabb-2a41-4bf7-ad50-dfda965b1081.png)
+
+### Link Column Properties
+
+Shown when a foreign key column or cell is selected.
+
+Link column properties include:
+
+- Column Name
+- Link Information
+  - Table source
+  - Column source
+  
+![image](/assets/design/specs/table_inspector/182350542-a5af060c-16f7-40b5-9a00-b7b6cc32e2a2.png)
+
+### Multiple Columns
+
+Shown when multiple columns are selected.
+
+Multiple column properties:
+
+- Data Type
+
+![image](/assets/design/specs/table_inspector/182350659-10d1b6f7-2cbf-44cb-a728-8241ad9f7335.png)
+
+### Actions Panel
+
+Shown at all times, the content might change depending on the selection status.
+
+The actions panel provides users with ready to use actions for quickly manipulating tables, columns or cells. The available actions are dependent on the selection.
+
+#### Future Considerations
+
+In the future I'm thinking the 'Actions' panel could include the data modeling suggestions. It could also include shortcuts to create queries that include the filters set in a table, or the group options into a summary.
+
+![image](/assets/design/specs/table_inspector/182186186-91449400-5608-42a2-8f05-33506cb2532b.png)
+
+### Selection Modes
+
+#### Cell Selection
+
+![image](/assets/design/specs/table_inspector/182121672-eaa8e422-7277-4421-8927-04637c182e6c.png)
+
+#### Column Selection
+
+![image](/assets/design/specs/table_inspector/182121417-8eb51f90-8767-4a01-bc36-a9c55c1614b5.png)
+
+#### Row Selection
+
+![image](/assets/design/specs/table_inspector/182122068-c08fecea-7d18-48ad-a8de-1cb1992ca8aa.png)
+
+#### To Deselect
+
+Deselect all selected items by clicking on a currently selected item, on an empty space or pressing the escape key.
```

Comment:
1. What is the initiation of the selection mode? Do you mean opening the inspector makes the table go into selection mode?
2. I think we should give a explicit action for `Deselect All`.